apps/emqx_exproto/src/emqx_exproto_channel.erl

```diff
@@ -576,17 +576,22 @@ default_conninfo(ConnInfo) ->
               receive_maximum => 0,
               expiry_interval => 0}.
 
-default_clientinfo(#{peername := {PeerHost, _},
-                     sockname := {_, SockPort}}) ->
+default_clientinfo(#{peername := {PeerHost, PeerPort},
+    sockname := {_, SockPort}}) when is_integer(PeerPort) ->
     #{zone         => external,
       protocol     => undefined,
       peerhost     => PeerHost,
+      peerport     => PeerPort,
```

Comment:
maybe add a small help function `peer_port/1` and call it here
so there is no need for another function clause for `default_clientinfo`

```
peer_port(Port) when is_integer(Port) -> Port;
peer_port(_) -> 0.
```
```suggestion
      peerport     => peer_port(PeerPort),
```

Reply:
and it applies to all other places where `peerport` is initialised.